app/code/community/Algolia/Algoliasearch/Helper/Entity/Producthelper.php

```diff
@@ -176,6 +176,7 @@ public function getProductCollectionQuery($storeId, $productIds = null, $only_vi
             }
 
             $products = $products
+                ->addAttributeToSelect('special_price')
```

Comment:
This was removed in ad4b976 but without it I couldn't see that the code was pulling in the special_prices for products anymore